### DISCUSSION TITLE
Stenographer dropping packets

### DISCUSSION THREAD

**Discussion by esher72** [Security-Onion-Solutions/securityonion#8651] [Unsupported Versions]
So I saw a similar ask in #5322 but it seems like it jumped right to a diagnosis of a disk IO issue. I have four sensors with 10Gb NICs peaking at about 2Gb/s but averaging right around 1Gb/s. Two of the nodes have recently started dropping packets in Steno, 50%+ at times. They are the two nodes under the heavier load and the drops do coincide with peak traffic. So it does seem to be a causal relationship between traffic volume and the drops. Overall the disk IO looks good though. NSM is on a large raid 6 disk with pretty fast disks. I do not see any errors logged that indicate any issues that could be related.Any suggestions on where to look next? I really appreciate any insight.

**Comment by dougburks** [comment]
Can you define exactly what you mean by `pretty fast disks`?

**Reply by esher72** [reply]
Specifically they are 8TB 7200RPM SAS drives in a 150TB RAID 6. Sorry for being vague and thanks for the response.

**Reply by dougburks** [reply]
I usually think of `pretty fast disks` as being NVME or SSD. 7200RPM drives are much slower. With that in mind, can you quantify what you mean by `Overall the disk IO looks good though`?

**Comment by mstannh** [comment]
We are also experiencing high % of stenographer packet loss (we also have a memory leak issue that might be related).
We have one sensor and in the last 24 Hours according to Grafana  we experienced a loss of  Max 98.5%	Mean 14.2%.

The Ubuntu MAN page for stenographer talks about the advantages of increasing 
 --blocks=NUM  
as well as adjusting 
  --preallocate_file_)mb=NUM

We configured an increased block size on the sensor .....and it was overwritten.... presumably by the manager node.
Apparently, updating the LOCAL version of the stenographer conf file has NO impact on the sensor. 
We are still looking for a fix, as we could not locate a conf file on the manager that is overwriting our stenographer conf .

**Reply by dougburks** [reply]
@mstannh From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.
